compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -2016,19 +2016,15 @@ void OMR::X86::CodeGenerator::doBinaryEncoding()
 
    // Create exception table entries for outlined instructions.
    //
-   auto oiIterator = self()->getOutlinedInstructionsList().begin();
-   while (oiIterator != self()->getOutlinedInstructionsList().end())
+   for(auto oiIterator = self()->getOutlinedInstructionsList().begin(); oiIterator != self()->getOutlinedInstructionsList().end(); ++oiIterator)
       {
       uint32_t startOffset = (*oiIterator)->getFirstInstruction()->getBinaryEncoding() - self()->getCodeStart();
       uint32_t endOffset   = (*oiIterator)->getAppendInstruction()->getBinaryEncoding() - self()->getCodeStart();
 
-      TR::Block * block = (*oiIterator)->getBlock();
-      bool      needsETE = (*oiIterator)->getCallNode() && (*oiIterator)->getCallNode()->getSymbolReference()->canCauseGC();
-
-      if (needsETE && block && !block->getExceptionSuccessors().empty())
+      auto block = (*oiIterator)->getBlock();
```

Comment:
I appreciate auto can be super helpful for things like iterators or other complex types - I don't know if I am a fan of loosing the type for simple things like block and node - surely the type helps make this code more clear?

Reply:
sure, I've updated it with ```TR::Block*``` and ```TR::Node*```.